cpp/src/parquet/page_index.h

```diff
@@ -17,7 +17,7 @@
 
 #pragma once
 
-#include "arrow/io/interfaces.h"
+#include "arrow/io/type_fwd.h"
```

Comment:
Why need this?

Reply:
`type_fwd.h` is a forward-declr for `interfaces`, here we might get compiling faster?